tests/unit/plugins/modules/service/test_service.py

```diff
@@ -0,0 +1,333 @@
+    }
+
+
+def test_missing_required(conn, module_args):
+    module_args(conn)
+
+    with pytest.raises(AnsibleFailJson, match="cluster, service"):
+        service.main()
+
+
+def test_missing_service(conn, module_args):
+    conn.update(service="example")
+    module_args(conn)
+
+    with pytest.raises(AnsibleFailJson, match="cluster"):
+        service.main()
+
+
+def test_missing_cluster(conn, module_args):
+    conn.update(cluster="example")
+    module_args(conn)
+
+    with pytest.raises(AnsibleFailJson, match="service"):
+        service.main()
+
+
+def test_present_invalid_cluster(conn, module_args):
+    conn.update(
+        cluster="example",
+        service="example",
+    )
+    module_args(conn)
+
+    with pytest.raises(AnsibleFailJson, match="Cluster does not exist"):
+        service.main()
+
+
+def test_present_missing_type(conn, module_args):
+    conn.update(
+        cluster=os.getenv("CM_CLUSTER"),
+        service=os.getenv("CM_SERVICE"),
+    )
+    module_args(conn)
+
+    with pytest.raises(AnsibleFailJson, match="type"):
+        service.main()
+
+
+def test_present_create_service(conn, module_args):
+    conn.update(
+        cluster=os.getenv("CM_CLUSTER"),
+        service=os.getenv("CM_SERVICE"),
+        display_name="Example Service",
+        type="ZOOKEEPER",
+    )
+    module_args(conn)
+
+    with pytest.raises(AnsibleExitJson) as e:
+        service.main()
+
+    assert e.value.changed == True
+
+    with pytest.raises(AnsibleExitJson) as e:
+        service.main()
+
+    assert e.value.changed == False
+
+
+def test_present_update_service(conn, module_args):
+    conn.update(
+        cluster=os.getenv("CM_CLUSTER"),
+        service=os.getenv("CM_SERVICE"),
+        display_name="Example Service by Ansible",
+    )
+    module_args(conn)
+
+    with pytest.raises(AnsibleExitJson) as e:
+        service.main()
+
+    assert e.value.changed == True
+
+    with pytest.raises(AnsibleExitJson) as e:
+        service.main()
+
+    assert e.value.changed == False
+
+
+def test_present_maintenance_mode(conn, module_args):
+    conn.update(
+        cluster=os.getenv("CM_CLUSTER"),
+        service=os.getenv("CM_SERVICE"),
+        maintenance="yes",
+    )
+    module_args(conn)
+
+    with pytest.raises(AnsibleExitJson) as e:
+        service.main()
+
+    assert e.value.service["maintenance_mode"] == True
+
+    with pytest.raises(AnsibleExitJson) as e:
+        service.main()
+
+    assert e.value.service["maintenance_mode"] == True
+    assert e.value.changed == False
+
+    conn.update(
+        maintenance="no",
+    )
+    module_args(conn)
+
+    with pytest.raises(AnsibleExitJson) as e:
+        service.main()
+
+    assert e.value.service["maintenance_mode"] == False
+
+    with pytest.raises(AnsibleExitJson) as e:
+        service.main()
+
+    assert e.value.service["maintenance_mode"] == False
+    assert e.value.changed == False
+
+
+def test_present_set_tags(conn, module_args):
```

Comment:
While testing I noticed following:
* If I set some tag manually before and I run this test it will set the desired tags but it will give back `AssertionError`, is this expected, shouldn't it just pass the test without  `AssertionError`.